tsconfig.json

```
@@ -0,0 +1,16 @@
+{
+  "compilerOptions": {
+    "declaration": true,
+    "experimentalDecorators": true,
+    "importHelpers": false,
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "removeComments": false,
+    "sourceMap": true,
+    "strict": true,
+    "target": "es5"
```

I would prefer that we target ES2017, so that async functions are preserved 🤔

I'm fairly certain that there are no ES2017 features missing in our supported browsers. At least `@babel/preset-env` doesn't pull in any transformations for the browsers we've selected. ES2018 does include object rest/spread though, which is not supported in our target browsers.
```suggestion
    "target": "ES2017"
```

---

> I'm fairly certain that there are no ES2017 features missing in our supported browsers.

This isn't for us alone—we've had a few requests for our _libraries_ to be ES5... so that's a thing.<sup>[\[1\]][1]</sup> I think it makes sense to have libraries ship the lowest common denominator so it doesn't require users (who might otherwise be writing ES5 code) to somehow compile their dependencies.

That said, we already don't do a great job of it, so it's possible that we document this breaking change where necessary and bump this up.

  [1]:https://github.com/search?q=org%3AMetaMask+ES5&type=Issues

---

We should ship _both_, which is what I did for [the onboarding library](https://github.com/MetaMask/metamask-onboarding/blob/master/rollup.config.js). But if we're only going to ship one, it should be the one that's most useful to us. Targeting ES5 means we get larger, less performant, and less debuggable code.

---

That fair, and where this these migrations will be accompanied by a major version bump, I'm fine with a higher target. We can add back a lower target as needed.

---

Sounds good! Maybe we won't need to... I have been seeing more and more ES6 libraries lately. The community is moving on, slowly.